api/src/main/java/graphql/nadel/hooks/HydrationArguments.java

```diff
@@ -0,0 +1,63 @@
+package graphql.nadel.hooks;
+
+import graphql.PublicApi;
+import graphql.language.Argument;
+import graphql.nadel.util.FpKit;
+import graphql.schema.GraphQLArgument;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@PublicApi
+public class HydrationArguments {
+
+    private final List<GraphQLArgument> hydrationRootGqlArguments;
+    private final List<Argument> hydrationRootAstArguments;
+
+    public HydrationArguments(List<GraphQLArgument> hydrationRootGqlArguments, List<Argument> hydrationRootAstArguments) {
+        if (hydrationRootGqlArguments == null || hydrationRootAstArguments == null) {
+            this.hydrationRootGqlArguments = Collections.emptyList();
+            this.hydrationRootAstArguments = Collections.emptyList();
+        } else {
+            this.hydrationRootGqlArguments = hydrationRootGqlArguments;
+            this.hydrationRootAstArguments = hydrationRootAstArguments;
+        }
+    }
+
+    public static HydrationArguments empty() {
+        return new HydrationArguments(Collections.emptyList(), Collections.emptyList());
+    }
+
```

Comment:
these are underlying schema `GraphQLArgument`, right?

Reply:
no, it's overall, see this test https://github.com/atlassian-labs/nadel/pull/194/files#diff-4d0ae59bc268469697cf0a858dd7d447c88b139fd3dbf646ccb2592a82e64ecbR1185